src/Plugin/GraphQL/DataProducer/Entity/EntityQuery.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace Drupal\graphql\Plugin\GraphQL\DataProducer\Entity;
+
+use Drupal\graphql\GraphQL\Execution\FieldContext;
+
+/**
+ * Builds and executes Drupal entity query.
+ *
+ * @DataProducer(
+ *   id = "entity_query",
+ *   name = @Translation("Load entities"),
+ *   description = @Translation("Loads entities."),
+ *   produces = @ContextDefinition("string",
+ *     label = @Translation("Entity IDs"),
+ *     multiple = TRUE
+ *   ),
+ *   consumes = {
+ *     "type" = @ContextDefinition("string",
+ *       label = @Translation("Entity type")
+ *     ),
+ *     "limit" = @ContextDefinition("integer",
+ *       label = @Translation("Limit"),
+ *       required = FALSE,
+ *       default_value = 10
+ *     ),
+ *     "offset" = @ContextDefinition("integer",
+ *       label = @Translation("Offset"),
+ *       required = FALSE,
+ *       default_value = 0
+ *     ),
```

Comment:
Offset based queries in Drupal are somewhat difficult in Drupal because Drupal allows access handlers that can not be put into a database query. This means that access filtering will be done after the fact which could mean that 10 entities are selected but none are actually sent back to the client.

For this same reason, generic "count" queries are not often seen in GraphQL schemas. On one hand because GraphQL is created for data sets where this isn't pratcial, on the other hand because access a single access check may make a count inaccurate.

Reply:
Yes, this can happen in some cases. I think we should add that to the docs as well.